src/Items/ItemHandler.h

```diff
@@ -154,18 +157,11 @@ class cItemHandler
 	Defaults to false unless overridden. */
 	virtual bool CanHarvestBlock(BLOCKTYPE a_BlockType);
 
-	static cItemHandler * GetItemHandler(int a_ItemType);
-	static cItemHandler * GetItemHandler(const cItem & a_Item) { return GetItemHandler(a_Item.m_ItemType); }
+	static cItemHandler * GetItemHandler(int a_ItemType, short a_ItemDamage);
 
-	static void Deinit();
-	
 protected:
 	int m_ItemType;
-	static cItemHandler * CreateItemHandler(int m_ItemType);
+	static void CreateItemHandler(int a_ItemType);
 
-	static cItemHandler * m_ItemHandler[E_ITEM_LAST + 1];
-	static bool m_HandlerInitialized;  // used to detect if the itemhandlers are initialized
+	static std::array<std::vector<std::unique_ptr<cItemHandler>>, E_ITEM_LAST + 1> m_ItemHandler;
```

Comment:
It would probably be useful to have a comment here about how this is basically a map of type -> { meta -> handler }.